### DISCUSSION TITLE
404 not getting while refresh the page

### DISCUSSION THREAD

**Discussion by ShaileshInfyom** [gatsbyjs/gatsby#31999] [Help]
## Description
Getting 404 while refresh the page
![image](https://user-images.githubusercontent.com/28335166/122563229-9418f200-d061-11eb-83f0-8b4cf7c2b1ed.png)

### Steps to reproduce
https://github.com/gatsbyjs/gatsby/tree/master/examples/simple-auth
I  download this code and create a build.
Then I created a virtual host in the wamp server and run the project.
now click on the profile link and open the page properly but I refresh the page and got 404. I getting the same issue on my website. so, I tried this demo but found the same issue.

Clear steps describing how to reproduce the issue. **Please** link to reproduction, this makes your issue _much_ easier to diagnose (seriously).

How to Make a Minimal Reproduction: https://www.gatsbyjs.com/contributing/how-to-make-a-reproducible-test-case/

### Expected result

The page should be refresh and open properly

### Actual result

Getting 404 Page not found

### Environment

  System:
    OS: Windows 10 10.0.19042
    CPU: (4) x64 Intel(R) Core(TM) i5-7400 CPU @ 3.00GHz
  Binaries:
    Node: 14.15.4 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.10 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 6.14.10 - C:\Program Files\nodejs\npm.CMD
  Languages:
    Python: 3.9.1 - /c/Users/GIGABYTE/AppData/Local/Programs/Python/Python39/python
  Browsers:
    Edge: Spartan (44.19041.1023.0), Chromium (91.0.864.48)
  npmPackages:
    gatsby: next => 3.9.0-next.0
    gatsby-plugin-react-helmet: next => 4.9.0-next.0
  npmGlobalPackages:
    gatsby-cli: 3.4.0

**Comment by LekoArts** [comment]
We can't give you specific help with WAMP so for that you'll need to reach out to other places like StackOverflow. If you want to preview your site you can use `gatsby serve` after a `gatsby build`.

**Comment by mitulgolakiya** [comment]
@LekoArts We have followed this same tutorial https://www.gatsbyjs.com/docs/deploying-to-digitalocean-droplet/ but with Apache, and we are facing the above same problem.

Is there any solution to that?

**Edit:** The URLs work fine on the localhost domain when we do `gatsby serve` but with `gatsby build` with a `public` folder with apache, these URLs not working.

**Comment by ShaileshInfyom** [comment]
here is workaround https://infyom.com/blog/fix-404-while-reloading-gatsby-website-for-dynamic-client-only-route
